### ISSUE TITLE
Code snippet won't play on iOS 15.5

### ISSUE THREAD

**Issue by pieterdd** [eshaz/icecast-metadata-js#206]
I wrote the following code snippet:

```html
<script src="https://eshaz.github.io/icecast-metadata-js/icecast-metadata-player-1.17.6.main.min.js"></script>
<script>
const player = new IcecastMetadataPlayer("https://ice4.somafm.com/groovesalad-64-aac", {
	onMetadata: (data) => {
  	document.querySelector('#metadata').innerHTML = data.StreamTitle;
  },
});
</script>
<div id="metadata">
metadata goes here
</div>
<button onclick="player.play()">
Play
</button>
```

It works fine on iOS 16.4:

![image](https://github.com/user-attachments/assets/1d807a37-6461-4ce7-945d-23944fb228d8)

It generates a CORS error on iOS 15.5 and refuses to play:

![image](https://github.com/user-attachments/assets/e594bdb2-38fb-4d98-a446-3f19a82dcc09)
![image](https://github.com/user-attachments/assets/05ad8a2f-e438-43d3-a1fc-bfb680e265e6)

Curiously, while https://eshaz.github.io/icecast-metadata-js/ too generates a CORS error on iOS 15.5 the stream does play sound:

![image](https://github.com/user-attachments/assets/35b126d9-2471-4d0a-abd8-7e95008e7421)

![image](https://github.com/user-attachments/assets/f34d3bfd-794a-4071-88f7-52e1e3c4ec2f)

I'm a bit puzzled by this, as I think the snippet is not that radically different from what the demo page does. Any idea why this code snippet is less lucky on iOS 15.5 than the official demo page?

**Comment by eshaz** [comment]
Your code snippet looks fine at first glance. It might be something to do with your development website your using or that specific version of iOS. I'll take a look.

For anyone interested in the background on this, I added the `Range` header by default a while back since a few Icecast server versions need this to return the full stream. See #198. This should not cause an issue since it's a [safelisted CORS header](https://developer.mozilla.org/en-US/docs/Glossary/CORS-safelisted_request_header). However, iOS Safari and desktop Safari choose not to follow the spec for a few versions and they enforce CORS for this header. I added a workaround for this bug to request the stream again without the `Range` header when it encounters this error. There might be some difference in the error that's preventing the workaround from triggering.

Ugg, add this one to the long list of problems, quirks, and bugs with iOS Safari...

**Comment by pieterdd** [comment]
Sorry to be the bringer of bad news :see_no_evil: Old browser compatibility indeed seems to be a pain in the neck.

If it helps, the demo plays without issue on iOS 15.5 (same stream):

![image](https://github.com/user-attachments/assets/a9ebac8c-1dad-4ec1-9841-8fe3b4bc7688)

So it appears that there a difference between how the demo loads the stream (which consistently works) and how my code snippet does it (which consistently doesn't).

**Comment by eshaz** [comment]
@pieterdd I have a hunch this may have something to do with codepen, since it appears that's where you're running the code according to your screenshots. Could you try your snippet using plain html / i.e. no codepen?

You may also try this out in the HTML demo page, which gives more control over the inputs.

There's nothing special in the demo; it just uses the IcecastMetadataPlayer class as documented.

**Comment by pieterdd** [comment]
@eshaz There's a test page up at https://somafm.com/ptest.html that triggered the same CORS error for me:

![image](https://github.com/user-attachments/assets/415d690d-af72-4c51-ad21-7c7416d7cd38)
![image](https://github.com/user-attachments/assets/eba70278-55a7-4b8f-95f5-3bbf35a0f6c2)

Granted, this "real device test" is running in a data center. But despite one CORS error the React demo page works fine:

![image](https://github.com/user-attachments/assets/cb7beb3d-c69a-4c1f-998e-e6353b7d1be4)
![image](https://github.com/user-attachments/assets/6ec74df2-9b7f-4dae-88f0-8ba702ece365)

**Comment by eshaz** [comment]
@pieterdd I released iceacast-metadata-player/1.17.7 that should resolve this issue. Please re-open if you notice it's still happening or if there are any regressions.

iOS Safari has / (had? apparently fixed in iOS 16) a very short timeout that allows audio / video to play after a user interaction. In this case, the time it took between the user interaction (clicking the play button) and the stream to load and start the audio playback was greater than this timeout. When this happens, an error event is emitted from the audio element. This should have been visible in the console when setting `enableLogging` in the IcecastMetadata player constructor options. `NotAllowedError: The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.` 

I already had a small chunk of silent audio in the code that is used to enable the play button on an audio element. I just used this to start the audio playback as soon as possible before the interaction timeout elapses.

**Comment by pieterdd** [comment]
Thanks for jumping through this hoop for an old version of Safari. I'm happy to confirm the fix works on my side!

![image](https://github.com/user-attachments/assets/cb6ee063-2b39-4907-a9c6-fd4127be5dfb)
